src/python/pants/engine/internals/platform_rules.py

```diff
@@ -28,18 +33,20 @@ def current_platform(env_tgt: EnvironmentTarget, global_options: GlobalOptions)
         # Else, it's a local environment.
         return Platform.create_for_localhost()
 
-    # Else, the environments mechanism is not used. For now, at least, we continue to support
-    # enabling remote execution globally for every build via the `--remote-execution` option.
-    return (
-        Platform.linux_x86_64
-        if global_options.remote_execution
-        else Platform.create_for_localhost()
-    )
+    # If the environments mechanism is not used, fall back to legacy behavior of
+    # `--remote-execution` being a global toggle.
+    if not environments_subsystem.names and global_options.remote_execution:
```

Comment:
This condition is repeated enough places that it might be good to make it a helper method on `EnvironmentsSubsystem` with a helper comment? It will help when/if we eventually deprecate this behavior.